qiskit_optimization/algorithms/warm_start_qaoa_optimizer.py

```diff
@@ -0,0 +1,339 @@
+                 circuit_factory: Optional[WarmStartQAOACircuitFactory] = None,
+                 aggregator: Optional[BaseAggregator] = None,
+                 penalty: Optional[float] = None,
+                 converters: Optional[Union[QuadraticProgramConverter,
+                                            List[QuadraticProgramConverter]]] = None
+                 ) -> None:
+        """ Initializes the optimizer. For correct initialization either
+            ``epsilon`` or ``circuit_factory`` must be passed. If only ``epsilon`` is specified
+            (either an explicit value or the default one), then an instance of
+            :class:`~qiskit.optimization.algorithms.WarmStartQAOACircuitFactory` is created.
+            If ``circuit_factory`` is specified then this instance is used in the implementation
+            and ``epsilon`` value is ignored.
+
+        Args:
+            pre_solver: An instance of an optimizer to solve the relaxed version of the problem.
+            relax_for_pre_solver: True if the problem must be relaxed to the continuous case
+                before passing it to the pre-solver.
+            qaoa: A QAOA instance to be used in the computations.
+            epsilon: the regularization parameter that changes the initial variables according to
+                xi = epsilon if xi < epsilon
+                xi = 1-epsilon if xi > epsilon.
+                The regularization parameter epsilon should be between 0 and 0.5. When it
+                is 0.5 then warm start corresponds to standard QAOA. If ``circuit_factory`` is
+                specified then this parameter is ignored. Default value is 0.25.
+            num_initial_solutions: An optional number of relaxed (continuous) solutions to use.
+                Default value is 1.
+            circuit_factory: An optional instance of the circuit factory to be used to create
+                circuits for the initial state and mixer. If None is specified then a default one,
+                an instance of :class:`~qiskit.optimization.algorithms.WarmStartQAOACircuitFactory`
+                is created using the value of the ``epsilon`` parameter.
+            aggregator: Class that aggregates different results. This is used if the pre-solver
+                returns several initial states.
+            penalty: The penalty factor to be used, or ``None`` for applying a default logic.
+            converters: The converters to use for converting a problem into a different form.
+                By default, when None is specified, an internally created instance of
+                :class:`~qiskit.optimization.converters.QuadraticProgramToQubo` will be used.
+
+        Raises:
+            QiskitOptimizationError: if ``epsilon`` is not in the range [0, 0.5].
+        """
+        if epsilon < 0. or epsilon > 0.5:
+            raise QiskitOptimizationError(
+                'Epsilon for warm-start QAOA needs to be between 0 and 0.5.')
```

Comment:
It would be nice to include the actual epsilon value that was received in the error message

Reply:
Included the actual value.